src/hardhat.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{fork::ForkSource, node::InMemoryNodeInner};
+use jsonrpc_core::{BoxFuture, Result};
+use jsonrpc_derive::rpc;
+use zksync_basic_types::{AccountTreeId, Address, U256};
+use zksync_core::api_server::web3::backend_jsonrpc::error::into_jsrpc_error;
+use zksync_types::{utils::storage_key_for_standard_token_balance, L2_ETH_TOKEN_ADDRESS};
+use zksync_utils::u256_to_h256;
+use zksync_web3_decl::error::Web3Error;
+
+/// Implementation of HardhatNamespaceImpl
+pub struct HardhatNamespaceImpl<S> {
+    node: Arc<RwLock<InMemoryNodeInner<S>>>,
+}
+
+impl<S> HardhatNamespaceImpl<S> {
+    /// Creates a new `Hardhat` instance with the given `node`.
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner<S>>>) -> Self {
+        Self { node }
+    }
+}
+
+#[rpc]
+pub trait HardhatNamespaceT {
+    #[rpc(name = "hardhat_setBalance")]
+    fn set_balance(
+        &self,
+        address: Address, // The address whose balance will be edited
+        balance: U256,    // The new balance to set for the given address, in wei
+    ) -> BoxFuture<Result<bool>>;
+}
+
+impl<S: Send + Sync + 'static + ForkSource + std::fmt::Debug> HardhatNamespaceT
+    for HardhatNamespaceImpl<S>
+{
+    /// Sets the balance of the given address to the given balance.
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The `Address` whose balance will be edited
+    /// * `balance` - The new balance to set for the given address, in wei
+    ///
+    /// # Returns
+    ///
+    /// A `BoxFuture` containing a `Result` with a `bool` representing the success of the operation.
+    fn set_balance(
+        &self,
+        address: Address,
+        balance: U256,
+    ) -> jsonrpc_core::BoxFuture<jsonrpc_core::Result<bool>> {
+        let inner = Arc::clone(&self.node);
+
+        Box::pin(async move {
+            let balance_key = storage_key_for_standard_token_balance(
```

Comment:
I believe you can use `storage_key_for_eth_balance(&address)` so you won't need to import `L2_ETH_TOKEN_ADDRESS`

Reply:
done